### PR TITLE
[Feature] Support ICNet

### PR DESCRIPTION
**Pull request by Junjun2016**
Support ICNet

- [x] Refactor
- [x] Add benchmark
- [x] Add unittests

### CODE REVIEW

**Comment by MengzhangLI** [comment]
Sources:
https://github.com/liminn/ICNet-pytorch
https://github.com/hszhao/ICNet
https://github.com/lxtGH/Fast_Seg

image

MMSegmentation:

image

**Comment by Junjun2016** [comment]
Can find more results [here](https://github.com/Junjun2016/ICNet).

File: configs/icnet/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Icnet for real-time semantic segmentation on high-resolution images
```
Comment (Junjun2016) [inline configs/icnet/README.md:1]:
```suggestion
# ICNet for real-time semantic segmentation on high-resolution images
```

File: configs/icnet/README.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+| ------ | ---------- | --------- | ------: | -------- | -------------- | ----: | ------------: | -------------------------------------------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| ICNet    | R-18-D8    | 832x832  |   80000 | 1.87      | 40.66           | 66.92 |         69.43 | [config](https://github.com/open-mmlab/mmsegmentation/blob/master/configs/icnet/icnet_r18-d8_832x832_80k_cityscapes.py)       | [model](https://download.openmmlab.com/mmsegmentation/v0.5/fcn/fcn_r50-d8_512x1024_40k_cityscapes/fcn_r50-d8_512x1024_40k_cityscapes_20200604_192608-efe53f0d.pth) &#124; [log](https://download.openmmlab.com/mmsegmentation/v0.5/fcn/fcn_r50-d8_512x1024_40k_cityscapes/fcn_r50-d8_512x1024_40k_cityscapes_20200604_192608.log.json)         |
```
Comment (Junjun2016) [inline configs/icnet/README.md:23]: Maybe we can Aevrage fps.

**Comment by MengzhangLI** [comment]
This PR is ended, please review new PR instead: https://github.com/open-mmlab/mmsegmentation/pull/884

**Pull request closed without merging**